### PR TITLE
Add index

### PR DESCRIPTION
**Pull request by cdupuis**
Add manually created `index.ts` to project.

I deliberately left out some internal functions and classes and not exported them. To make this clear, I created empty exports for internal files in the `index.ts` to not mistakenly think they were just forgotten to get exported. 

Not sure if that covers every use-case so far in skills, but it is a start IMHO. Interested in your feedback @ddgenome and @ipcrm.

I've migrated a skill to use the index: https://github.com/atomist-skills/github-auto-merge-skill/pull/9

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    Contextual,
+    CommandContext,
+    CommandHandler,
+    Configuration,
+    EventContext,
+    EventHandler,
+    HandlerStatus,
+} from "./lib/handler";
+export { HttpClient } from "./lib/http";
+export { debug, error, info, warn } from "./lib/log";
```
Comment (ddgenome) [inline index.ts:41]: Could/should these all be under one `log` export and used like `log.debug`, `log.error`, etc.?

File: index.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    slackErrorMessage,
+    slackFooter,
+    slackInfoMessage,
+    slackSeparator,
+    slackSupportLink,
+    slackSuccessMessage,
+    slackTs,
+    slackQuestionMessage,
+    slackWarningMessage,
+} from "./lib/messages";
```
Comment (ddgenome) [inline index.ts:60]: Could/should these all be available under a single `slack` or `slackMessage` export?

File: index.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+export {
+    ParameterPrompt,
+    ParametersPromptObject,
```
Comment (ddgenome) [inline index.ts:63]: Why is this one plural?

File: index.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    ParameterPromptOptions,
+    ParameterPromptStyle,
+} from "./lib/parameter_prompt";
```
Comment (ddgenome) [inline index.ts:66]: Exported under `prompt`?

File: index.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+export { BaseParameter, Option, Options, Parameter, ParametersObjectValue, HasDefaultValue } from "./lib/parameters";
```
Comment (ddgenome) [inline index.ts:67]: Export under `parameter`?

File: index.ts [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+export {} from "./lib/payload";
+export {
+    ProjectLoader,
+    CloneOptions,
+    AuthenticatedRepositoryId,
+    gitHubComRepository,
+    RepositoryId,
+    RepositoryProviderType,
+} from "./lib/project";
```
Comment (ddgenome) [inline index.ts:76]: Export under `project`?

**Comment by ddgenome** [comment]
This index provides everything used by the k8s pod health skill.

**Comment by cdupuis** [comment]
Thanks @ddgenome. I'll go ahead and merge this and we can iterate in master.

**Comment by cdupuis** [comment]
Opps, hang on. I'm seeing your comments now. Addressing them.

File: index.ts (hunk 1 above)
Comment (cdupuis) [reply index.ts:41]: Done.

**Comment by ddgenome** [comment]
They weren't recommendations, just talking points/things to consider.

**Comment by cdupuis** [comment]
Closing in favour of #14

**Pull request closed without merging**